admin/class-admin-asset-manager.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * This class registers all the necessary styles and scripts. Also has methods for the enqueing of scripts and styles. It automatically adds a prefix to the handle.
+ */
+class WPSEO_Admin_Asset_Manager {
+
+
+	/**
+	 *  Prefix for naming the assets.
+	 */
+	const PREFIX = 'yoast-seo-';
+
+	/**
+	 * Enqueues scripts.
+	 *
+	 * @param string $script The name of the script to enqueue.
+	 */
+	public function enqueue_script ( $script ) {
+		wp_enqueue_script( self::PREFIX . $script );
+	}
+
+	/**
+	 * Enqueues styles.
+	 *
+	 * @param string $style The name of the style to enqueue.
+	 */
+	public function enqueue_style ($style) {
+		wp_enqueue_style( self::PREFIX . $style );
+	}
+
+	/**
+	 * Calls the functions that register admin-scripts and admin-styles.
+	 */
+	public function register_assets() {
+		$this->register_scripts();
+		$this->register_styles();
+	}
+
+	/**
+	 * Registers admin-scripts. Can be enqueued when they are needed.
+	 */
+	protected function register_scripts() {
```

Comment:
I suggest storing the variable paramaters of each script (and style) in an array. Then you can simply loop trough these arrays and call the `wp_register_script` function while using the stored data. This makes it a bit more readable and maintainable.